src/tests/ftest/launch.py

```diff
@@ -1482,7 +1482,13 @@ def process_the_cores(avocado_logs_dir, test_yaml, args):
                                       "quit"])
 
             last_line = gdb_output.splitlines()[-1]
-            exe_name = last_line[7:last_line[7:].find(" ") + 7]
+            cmd = last_line[7:-1]
+            # assume there are no arguments on cmd
+            find_char = "'"
+            if cmd.find(" ") > -1:
+                # there are arguments on cmd
+                find_char = " "
```

Comment:
Optional: Apparently `in` is faster than `find()` and if you wanted to save some space:
```suggestion
            find_char = " " if " " in cmd else "'"
```

Reply:
Ah, yes.  I forget about python's ability to use `in` to find things in strings.  I'll plop that change into another upcoming PR if you don't mind.